src/component/Autocomplete.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+export interface AutocompleteProps {
+  suggestions: string[];
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = ({ suggestions }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;
+    setInputValue(userInput);
+
+    const filtered = suggestions.filter(
+      suggestion => suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+    );
+    setFilteredSuggestions(filtered);
+    setShowSuggestions(true);
+  };
+
+  const handleClick = (suggestion: string) => {
+    setInputValue(suggestion);
+    setFilteredSuggestions([]);
+    setShowSuggestions(false);
+  };

```

Comment:
Refer this document while handling batch updates for multiple state values

https://react.dev/learn/queueing-a-series-of-state-updates

Reply:
kk sir